public/app/features/dropandpaste/DashboardDropOverlay.tsx

```diff
@@ -0,0 +1,169 @@
+import { css, keyframes } from '@emotion/css';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Trans } from '@grafana/i18n';
+import { Icon, Stack, useStyles2, Text } from '@grafana/ui';
+
+export function DashboardDropOverlay() {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div className={styles.dropOverlay}>
+      <div className={styles.dropHint}>
+        <Icon name="upload" size="xxl" className={styles.dropIcon} />
+        <Text variant="body" element="p">
+          <Trans i18nKey="dragndrop.drop-hover-text">Drop file to upload it to your dashboard</Trans>
+        </Text>
+        <Stack gap={3}>
+          <JsonIcon />
+          <XlsIcon />
+          <ImgIcon />
+          <TxtIcon />
+        </Stack>
+      </div>
+    </div>
+  );
+}
+
+function getStyles(theme: GrafanaTheme2) {
+  return {
+    dropOverlay: css({
+      display: 'flex',
+      backdropFilter: 'blur(2px)',
+      position: 'absolute',
+      zIndex: theme.zIndex.modal,
+      inset: 0,
+      alignItems: 'center',
+      justifyContent: 'center',
+      paddingBottom: theme.spacing(10),
+    }),
+    dropHint: css({
+      alignItems: 'center',
+      display: 'flex',
+      flexDirection: 'column',
+      gap: theme.spacing(1),
+      backgroundColor: theme.colors.background.secondary,
+      border: `1px dashed ${theme.colors.border.weak}`,
+      borderRadius: theme.shape.radius.default,
+      padding: theme.spacing(2, 4),
+    }),
+    dropIcon: css({
+      [theme.transitions.handleMotion('no-preference')]: {
+        animationName: bounce,
+        animationIterationCount: 'infinite',
+        animationDuration: '0.6s',
+        animationTimingFunction: 'ease-out',
+      },
+    }),
+  };
+}
+
+const bounce = keyframes({
+  'from, to': {
+    transform: 'translateY(0px)',
+  },
+  '50%': {
+    transform: 'translateY(-8px)',
+  },
+});
+
+function ImgIcon() {
```

Comment:
Nit: Maybe these svg funcs could live outside the component to keep this file cleaner?